### PR TITLE
Feature IM-13 PDF Abstraktion

### PR DESCRIPTION
**Pull request by tilokowalski**

### CODE REVIEW

File: src/main/java/com/fes/hazelnut/schoolnut/document/helper/HeaderFooterHandler.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+import com.itextpdf.text.pdf.ColumnText;
+import com.itextpdf.text.pdf.PdfContentByte;
+import com.itextpdf.text.pdf.PdfGState;
+import com.itextpdf.text.pdf.PdfPageEventHelper;
+import com.itextpdf.text.pdf.PdfWriter;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.Base64;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/helper/HeaderFooterHandler.java:22]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'org.apache.commons.lang3.exception.ExceptionUtils'

File: src/main/java/com/fes/hazelnut/schoolnut/data/entity/config/ConfigEntity.java [hunk 2]
```diff
@@ -1,7 +1,12 @@
 package com.fes.hazelnut.schoolnut.data.entity.config;
 
 import com.fes.hazelnut.schoolnut.data.entity.EntityMutuals;
+
+import jakarta.persistence.Column;
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/data/entity/config/ConfigEntity.java:5]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'jakarta.persistence.Column'

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/ClasslistDocument.java [hunk 3]
```diff
@@ -4,15 +4,39 @@
...
 import com.fes.hazelnut.schoolnut.document.Document;
 import com.itextpdf.text.DocumentException;
+import com.itextpdf.text.Font;
+import com.itextpdf.text.FontFactory;
 import com.itextpdf.text.Paragraph;
+import com.itextpdf.text.Phrase;
+import com.itextpdf.text.Rectangle;
+import com.itextpdf.text.pdf.PdfPCell;
+import com.itextpdf.text.pdf.PdfPTable;
+import com.itextpdf.text.pdf.PdfDiv.BorderTopStyle;
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/ClasslistDocument.java:14]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'com.itextpdf.text.pdf.PdfDiv.BorderTopStyle' import. Should be before 'com.itextpdf.text.pdf.PdfPTable'.

File: src/main/java/com/fes/hazelnut/schoolnut/document/pdf/ClasslistDocument.java [hunk 4]
```diff
@@ -4,15 +4,39 @@
...
 import java.io.FileNotFoundException;
 import java.io.IOException;
 
+import org.w3c.dom.css.Rect;
```
Comment (tilokowalski) [inline src/main/java/com/fes/hazelnut/schoolnut/document/pdf/ClasslistDocument.java:18]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'org.w3c.dom.css.Rect'

**Review by VogtJoshua — DISMISSED** [review]

**Review by thisami — APPROVED** [review]

**Review by thisami — APPROVED** [review]

**Review by VogtJoshua — APPROVED** [review]

**Review by lenav99 — APPROVED** [review]

**Pull request merged**